q2_moshpit/busco/utils.py

```diff
@@ -0,0 +1,384 @@
+    appending a '--' prefix and concatenating words separated by a '_',
+    e.g.: 'some_parameter_x' -> '--someParameterX'.
+
+    Args:
+        arg_key (str): Argument name.
+        arg_val: Argument value.
+
+    Returns:
+        [converted_arg, arg_value]: List containing a prepared command line
+            parameter and, optionally, its value.
+    """
+
+    # If the key is one of
+    if arg_key in arguments_with_hyphens.keys():
+        arg_key = arguments_with_hyphens[arg_key]
+
+    if isinstance(arg_val, bool) and arg_val:
+        return [f"--{arg_key}"]
+    else:
+        return [f"--{arg_key}", str(arg_val)]
+
+
+def _draw_busco_plots_for_render(
+    df: pd.core.frame.DataFrame,
+    width: int = None,
+    height: int = None,
+    labelFontSize: int = None,
+    titleFontSize: int = None,
+) -> str:
+    """
+    Draws a hroizontal normalized bar plot for every sample for which BUSCO was
+    run. Each barplot shows the BUSCO results for each of the MAGs in the
+    sample. The plots for all samples are drwan in one composite plot which
+    is then returned as a dictionary for rendering (but casted to a string).
+
+    Args:
+        df (pd.core.frame.DataFrame): tabular batch summary for all samples
+        width (int): width of the plot
+        height (int): height of the plot
+        labelFontSize (int): size of the labels in plot
+        titleFontSize (int): size of titles in plot
+
+    Output:
+        Output plot in dictionary from casted to a string.
+    """
+
+    # Clean column names
+    df.columns = df.columns.str.replace(" ", "_")
+    df.columns = df.columns.str.replace("[^a-zA-Z0-9_]", "", regex=True)
+    df.columns = df.columns.str.lower()
+
+    # Rename column "input_file"
+    df["mag_id"] = df["input_file"].str.split(".", expand=True)[0]
+
+    # Cast into percent_gaps col to float
+    df["percent_gaps"] = df["percent_gaps"].str.split(
+        '%', expand=True
+    )[0].map(float)
+
+    # Get number of samples
+    n_samples = len(df["mag_id"].unique())
+
+    # Format data for plotting
+    busco_plot_data = pd.melt(
+        df,
+        id_vars=["sample_id", "mag_id", "dataset", "n_markers"],
+        value_vars=["single", "duplicated", "fragmented", "missing"],
+        value_name="BUSCO_percentage",
+        var_name="category",
+    )
+
+    secondary_plot_data = df[[
+        "sample_id",
+        "mag_id",
+        'scaffold_n50',
+        'contigs_n50',
+        'percent_gaps',
+        'number_of_scaffolds',
+    ]]
+
+    # Specify order
+    mapping = {"single": 1, "duplicated": 2, "fragmented": 3, "missing": 4}
+    busco_plot_data["order"] = busco_plot_data["category"].map(mapping)
+
+    # Estimate fraction of sequences in each BUSCO category
+    busco_plot_data["fracc_markers"] = (
+        "~"
+        + round(
+            busco_plot_data["BUSCO_percentage"] *
+            busco_plot_data["n_markers"] / 100
+        ).map(int).map(str)
```

Comment:
The round function returns a column of floats when the input is a pandas column, e.g. 3.0 or so. In order to get rid of the .0 I cast it to an int. And I round before casting to int because just casting to int doesn't round, it simply returns the the preceding integer of int input.  

Example:

```
round(df["FloatColumn"])
0    1.0
1    3.0
2    6.0
3    8.0
4    9.0
```